tool_collections/bamtools/bamtools_filter/bamtools-filter.xml

```diff
@@ -1,322 +1,299 @@
-filtering will look like this::
-
- !(1) &amp; (2 | 3)
-
-Here, numbers 1, 2, and 3 represent conditions. The following screenshot illustrates how to do this in Galaxy:
-
-.. image:: rule.png
-
-There are three conditions here, each with a single filter. A text entry area that can be opened by clicking on the **Would you like to set rules?** checkbox enables you to enter a rule.
-Here numbers correspond to numbers of conditions as they are shown in the interface. E.g., 1 corresponds to condition 1, 2 to condition 2 and so on... In human language this means::
-
- NOT condition 1 AND (condition 2 OR condition 3)
-
------
-
-**JSON script file**
-
-This tool produces two outputs. One of the them is a BAM file containing filtered reads. The other is a JSONified script. It can help you to see how your instructions are sent to BAMTools.
-For instance, the example 4 looks like this in the JSON form::
-
-       {
-        "filters":
-        [
-          { "id": "1",
-            "tag":"NM:=0",
-            "isReverseStrand":"false"
-          },
-          { "id": "2",
-            "tag":"NM:>0",
-            "isReverseStrand":"true"
-          }
-        ]
-      }
-
-
------
-
-**More information**
-
-.. class:: infomark
-
-Additional information about BAMtools can be found at https://github.com/pezmaster31/bamtools/wiki
-
-
-    </help>
-    <citations>
-        <citation type="doi">10.1093/bioinformatics/btr174</citation>
-    </citations>
-</tool>
+<tool id="bamFilter" name="Filter" version="2.4.1">
+    <description>BAM datasets on a variety of attributes</description>
+    <requirements>
+        <requirement type="package" version="2.4.0">bamtools</requirement>
+    </requirements>
+    <command>
+        <![CDATA[
+            cat $script_file > $out_file2;
+            ln -s "${input_bams}" "localbam.bam" &&
+            ln -s "${input_bams.metadata.bam_index}" "localbam.bam.bai" &&
+            bamtools filter -script $script_file -in "localbam.bam" -out "$out_file1"
+        ]]>
+    </command>
+    <configfiles>
+        <configfile name="script_file">
+<![CDATA[
+<% import json %>
+#set $config = dict()
+#if $conditions:
+    #set $config[ 'filters' ] = []
+    #for $i, $condition in enumerate( $conditions, start=1 ):
+        #set $filter = dict( id=str( $i ) )
+        #for $j, $s in enumerate( $condition.filters, start=1 ):
+            #set $filter[ $str( s[ 'bam_property' ][ 'bam_property_selector' ] ) ] = $str( s[ 'bam_property' ][ 'bam_property_value' ] )
+        #end for
+        $(config[ 'filters' ].append( $filter ))
+    #end for
+    #if str( $rule_configuration.rules_selector ) == "True":
+        #set $config[ 'rule' ] = $rule_configuration.rules
+    #end if
+#end if
+$json.dumps( $config, indent=4 )
+]]>
+        </configfile>
+    </configfiles>
+    <inputs>
+        <param name="input_bams" type="data" format="bam" label="BAM dataset(s) to filter" />
```

Comment:
I think we should remove `(s)` in the label and maybe also change the name to `input_bam`.